atlp-rwanda/script-heroes-bn-backend
# #174797534 Add Search Functionality

### What does this PR do?
Allow users to search for requests

### Description of Task to be completed?

- Pull the repo to get the project's latest version
- Create a branch to work on
- Rebase
- Create an endpoint that takes a given search in params
- Create controller for searching requests
- Run `npm run dev`
- Write Tests
- Add, commit and push changes

### How should this be manually tested?

1. Clone the repo `git clone https://github.com/atlp-rwanda/script-heroes-bn-backend.git`
2. Switch your working directory `cd script-heroes-bn-backend`
3. Checkout the branch git checkout `ft-search-functionality-#174797534`
4. Install all dependencies `npm install`
5. Run tests `npm run dev`
6. Login on `/api/auth/login`
7. Make sure you have some trips already created in DB
8. Search Trips by: 
     - Start date `/api/allTripsByStartDate/search?startDate=date`
     - End date `/api/allTripsByEndDate/search?endDate=date`
     - Origin `/api/allTripsByOrigin/search?origin=countryFrom`
     - Destination `/api/allTripsByDest/search?destination=countryTo`
     - Both Origin & Destination `/api/allTripsByOriginDest/search?origin=countryFrom&destination=countryTo`
9. Search Requests by: 
     - Status `/api/allRequestsByStatus/search?status=status` (status = pending/approved/rejected)
     - id `/api/getRequestById/search?id=requestId`
     - user id `/api/getRequestByUser/search?id=userId`
10. Also run `npm run test`

<img width="960" alt="Screenshot (93)" src="https://user-images.githubusercontent.com/49404485/97994459-00eebd00-1dee-11eb-8bea-11cdfcb543f4.png">


### What are the relevant pivotal tracker stories?

[#174797534](https://www.pivotaltracker.com/n/projects/2465784/stories/174797534)
